### ISSUE TITLE
Fix S2583 FP: Doesn't follow try-catch in loop

### ISSUE THREAD

**Issue by Piedone** [SonarSource/sonar-dotnet#8736]
### Description

[S2583 Conditionally executed code should be reachable](https://rules.sonarsource.com/csharp/RSPEC-2583/) shows a violation where there really isn't, perhaps because it doesn't properly follow a try-catch construct inside a `while` loop.

### Repro steps

```c#
var shouldContinue = true;
var index = 0;
while (shouldContinue)
{
    try
    {
        await Console.Out.WriteLineAsync("dummy");
        shouldContinue = false;
    }
    catch (Exception)
    {
        index++;

        if (index == 5)
        {
        }
    }
}
```

### Expected behavior

There is no violation, since if `Console.Out.WriteLineAsync)` throws an exception, `index` will be incremented, and the loop body rerun. If this happens five times, `index` will reach `5` and the `if` condition will be `true`.

### Actual behavior

Violation on line `if (index == 5)`.

### Known workarounds

Haven't found any.

### Related information

* C#/VB.NET Plugins version: we use the NuGet package v9.19.0.84025.
* Visual Studio version: 17.8.6.
* MSBuild / dotnet version: 8.0.101.
* SonarScanner for .NET version (if used): not used.
* Operating System: Windows 11 Pro v23H2 build 22631.3007

**Comment by Tim-Pohlmann** [comment]
Hello @Piedone thanks for the report. I confirm this as an FP. Fortunately, this will be fixed with the release of 9.20! See https://github.com/SonarSource/sonar-dotnet/issues/8474

**Comment by Piedone** [comment]
Great, thanks!
